aospy/automate.py

```
@@ -0,0 +1,220 @@
+"""Functionality for specifying and cycling through multiple calculations."""
+from __future__ import print_function
+
+import itertools
+import logging
+
+from .calc import Calc, CalcInterface
+
+try:
+    import colorama
+except ImportError:
+    pass
+try:
+    import multiprocess
+except ImportError:
+    pass
+
+
+def _get_obj_default_or_all(name, parent, attr_name, defaults_attr_name):
+    if name == 'all':
+        return getattr(parent, attr_name).values()
+    elif name == 'default':
+        return getattr(parent, defaults_attr_name).values()
+    else:
+        return getattr(parent, attr_name)[name]
```

Not necessarily opposed, just want to confirm:  does this mean that we need to switch from using lists to dictionaries when we specify the ```models``` attribute of a ```Proj``` object or the ```runs``` attribute of a ```Model``` object?

E.g. go from this:
```python
example_model = Model(
    name='example_model',
    grid_file_paths=('/path/to/files'),
    runs=[runs.control, runs.conv_off],
    default_runs=[]
)
```
to this:
```python
example_model = Model(
    name='example_model',
    grid_file_paths=('/path/to/files'),
    runs=dict(control=runs.control, conv_off=runs.conv_off),
    default_runs=dict()
)
```

I'm assuming this was a decision made to eliminate the need for the ```find_obj.py``` module?  (I'm all for that decision, by the way)

---

This is actually already taken care of via the [`utils.io.dict_name_keys`](https://github.com/spencerahill/aospy/blob/develop/aospy/model.py#L194) function.  So the list syntax still works.

---

On that point, the main script and example object library from these commits should work; try running the main script (from inside the example directory so that you don't have to install the object library).

---

Ah I see this gets addressed automatically, should have noticed that sorry -- should we add some checking to ```utils.io.dict_name_keys``` to make sure that there are no duplicate names in the list of provided core objects (at least warn, maybe raise an exception)?  

For example, right now if you had two Runs with the same name attribute in the list provided, the first one would get overwritten by the second without the user knowing. 

> On that point, the main script and example object library from these commits should work; try running the main script (from inside the example directory so that you don't have to install the object library).

Indeed, I can confirm this works!